### PR TITLE
[MIG] Migrate to V9 and add different sequences for customers and vendors

### PR DESCRIPTION
**Pull request by mncoon**
All other created partners don't receive a sequence number.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Hi, @copmn, thanks for the contribution. Have you read the procedure https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0 to check if there's any pending update from 8.0 branch?

About the PR itself, please squash the 2 commits in one, as they have even the same commit message. More comments inline.

File: base_partner_sequence/README.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+when you create a new partner, using the defined sequence.
+
+The reference field is treated as a commercial field, i.e. it
+is managed from the commercial partner and then propagated to
+the partner's contacts. The field is visible on the contacts,
+but it can only be modified from the commercial partner.
+
+No references are assigned for contacts such as shipping and
+invoice addresses.
+This module is a migration of the original base_partner_sequence
```
Comment (pedrobaeza) [inline base_partner_sequence/README.rst:20]: This text can be removed, as it's deprecated.

File: base_partner_sequence/README.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+addon to OpenERP version 7.0.
+
+
+Usage
+=====
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/134/8.0
```
Comment (pedrobaeza) [inline base_partner_sequence/README.rst:29]: Change here to 9.0

File: base_partner_sequence/README.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/partner-contact/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/partner-contact/issues/new?body=module:%base_partner_sequence%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (pedrobaeza) [inline base_partner_sequence/README.rst:38]: 9.0

File: base_partner_sequence/__init__.py [hunk 4]
```diff
@@ -1,22 +1,3 @@
 # -*- encoding: utf-8 -*-
```
Comment (pedrobaeza) [inline base_partner_sequence/__init__.py:1]: s/encoding/coding

File: base_partner_sequence/__openerp__.py [hunk 5]
```diff
@@ -21,34 +21,19 @@
 #
 ##############################################################################
```
Comment (pedrobaeza) [inline base_partner_sequence/__openerp__.py:22]: Please use short headers like in https://github.com/OCA/maintainer-tools/blob/master/template/module/__init__.py

File: base_partner_sequence/models/partner.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    def _commercial_fields(self):
+        """
+        Make the partner reference a field that is propagated
+        to the partner's contacts
+        """
+        return super(ResPartner, self)._commercial_fields() + ['ref']
+
+    @api.model
+    def _get_next_ref(self, partner=None, vals=None):
+        if vals['customer'] == True:
```
Comment (pedrobaeza) [inline base_partner_sequence/models/partner.py:44]: You can put only `if val['customer']:`. It works the same

**Comment by pedrobaeza** [comment]
The problem I see with your change of approach is that the separated sequence numbers that you have put for customers and vendors doesn't fit the current data model. You have partners, and these partners can be customers, vendors, or both, so it doesn't make sense to give a number for a customer that later can be also a supplier.

Please remove this new feature, although the contact not being given a number is good.

File: base_partner_sequence/README.rst (hunk 3 above)
Comment (yvaucher) [inline base_partner_sequence/README.rst:38]: You can remove the link here. The template has been simplyfied: https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

File: base_partner_sequence/README.rst [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+Credits
+=======
+
+Contributors
+------------
+
+* Thomas Rehn <thomas.rehn@initos.com>
+* Stefan Rijnhart <stefan@therp.nl>
+* Yannick Vaucher <yannick.vaucher@camptocamp.com>
```
Comment (yvaucher) [inline base_partner_sequence/README.rst:49]: Please remove me from credits, I just ran a script to change the icon of this module to default OCA icon.

File: base_partner_sequence/models/partner.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+from openerp import models, api
+
+
+class ResPartner(models.Model):
+    """Assigns 'ref' from a sequence on creation and copying"""
+
+    _inherit = 'res.partner'
+
+    @api.model
+    def _needsRef(self, partner_id=None, vals=None):
```
Comment (yvaucher) [inline base_partner_sequence/models/partner.py:15]: Why is this the only method to be camel case ?

File: base_partner_sequence/models/partner.py [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    @api.model
+    def create(self, vals):
+        if not vals.get('ref') and self._needsRef(vals=vals):
+            vals['ref'] = self._get_next_ref(vals=vals)
+        return super(ResPartner, self).create(vals)
+
+    @api.multi
+    def write(self, vals):
+        for partner in self:
+            ref = vals.get('ref') if 'ref' in vals else partner.ref
```
Comment (yvaucher) [inline base_partner_sequence/models/partner.py:58]:
```
ref = vals.get('ref', partner.ref)
```

does the same

File: base_partner_sequence/models/partner.py [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+            if not ref and self._needsRef(partner.id, vals):
+                vals['ref'] = self._get_next_ref(partner, vals)
+            super(ResPartner, partner).write(vals)
+        return True
+
+    @api.one
```
Comment (yvaucher) [inline base_partner_sequence/models/partner.py:64]: api.one is deprecated, please use `api.multi` instead

**Comment by rafaelbn** [comment]
Hi @mncoon , thanks for this PR, Please could you review comments?

**Comment by pedrobaeza** [comment]
@mncoon, are you going to handle comments?

**Comment by pedrobaeza** [comment]
Superseeded by #358

**Pull request closed without merging**